pkgs/development/libraries/intel-media-sdk/default.nix

```diff
@@ -27,6 +27,11 @@ stdenv.mkDerivation rec {
   ];
   nativeCheckInputs = [ gtest ];
 
+  # For forward compatibility with oneVPL.
+  # To ensure the VPL backend is dispatched at runtime, export INTEL_MEDIA_RUNTIME=ONEVPL
+  # See https://github.com/Intel-Media-SDK/MediaSDK#media-sdk-support-matrix
+  runtimeInputs = [ libvpl ];
```

Comment:
Can we use patchelf instead of LD_LIBRARY_PATH?

Reply:
oh good point, I can try that

Reply:
I keep coming back to this, hoping more of staging is present in cache.nixos.org, otherwise dependencies are failing to build for me... today it's progressing farther.

my change looks like this, removing runtimeInputs:
```nix
  postFixup = ''
    patchelf --add-needed ${libvpl}/lib/libvpl.so
  '';
```

I don't remember where I found runtimeInputs, and it doesn't seem documented. What is the actual difference between runtimeInputs and patchelf?

Reply:
Staging is intentionally not cached or even built for the most part.

My recommendation always is to base your staging PRs on the merge-base between master and staging. That way you can test things with the cache of master for everything but the thing you're touching and it should still apply cleanly to staging.

I don't remember the exact flag name but I think this would technically be better as it'd allow grafting:

```Nix
  postFixup = ''
    patchelf --add-needed libvpl.so --add-rpath ${libvpl}/lib/libvpl.so
  '';
```

Reply:
reading the intel-media-sdk compatibility chart again I'm pretty sure I needed to patchelf onevpl-intel-gpu's libmfx-gen.so onto intel-media-sdk's libmfx.so so it can find it at runtime if directed. [This line in intel-media-sdk](https://github.com/Intel-Media-SDK/MediaSDK/blob/7a72de33a15d6e7cdb842b12b901a003f7154f0a/api/mfx_dispatch/linux/mfxloader.cpp#L194) is clearly trying to load libmfx-gen.so which is not provided by libvpl, but by onevpl-intel-gpu.

@Atemu This raises the closure size of intel-media-sdk (and ffmpeg, if withMfx or withVpl are set) by 10.6MB.

I'm not exactly sure how to test this fallback mechanism in isolation, but I've tried the following with success:
1. use this PR's intel-media-sdk and ffmpeg-full (with no overrides)
2. use ffmpeg to encode using a qsv backend without `INTEL_MEDIA_RUNTIME` set
3. use ffmpeg to encode using a qsv backend with `INTEL_MEDIA_RUNTIME=ONEVPL`
4. check with `intel_gpu_top` that it is in fact using the gpu for both encodes

Reply:
using patchelf in the latest revision, btw

after looking more, I don't think `runtimeInputs` is even read by `mkDerivation`

Reply:
I've removed the patch for intel-media-sdk based on my reasoning above. I think if there's a clear need for it, it can easily be added back in another PR to make incremental improvement

Reply:
> EDIT 2: using libvpl from #291559 and `hardware.opengl.extraPackages = [ pkgs.onevpl-intel-gpu ];` I can use this PR's ffmpeg-full `withVpl = true;` with an **unmodified intel-media-sdk**.

This is how it should work. `onevpl-intel-gpu` is a driver package and should be installed to the global driver path by the user.

That's how all other hardware-dependant accelerator APIs work in Nixpkgs.